.github/workflows/tests_env.yml

```diff
@@ -0,0 +1,61 @@
+name: Tests familly project
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Which version, empty for development, `release` for latest released version'
+        default: ""
+
+  pull_request:
+
+defaults:
+  run:
+    # The slightly odd shell call is to force bash to read .bashrc, which is
+    # necessary for having conda behave sensibly.
+    shell: bash -l -e {0}
+
+jobs:
+  tests:
+    name: ${{ matrix.repo }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        repo: [qutip-qip, qutip-jax, qutip-qoc, qutip-qtrl]
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: conda-incubator/setup-miniconda@v3
+        with:
+          auto-update-conda: true
+          python-version: 3.11
+
+      - name: Install QuTiP and dependencies
+        run: |
+          # Install the extra requirement
+          python -m pip install pytest pytest-rerunfailures  # tests
```

Comment:
What do we need pytest-rerunfailures for?

Reply:
It's a requirement for tests in `setup.cfg`.
It add the pytest mark `flaky` used for metric and simdiag tests.

Reply:
Ah. It would be nice to remove those eventually.